### ISSUE TITLE
💅 False warning on separator role

### ISSUE THREAD

**Issue by astanciu** [biomejs/biome#7076]
`<div role="separator">`

throws this error:
The element with the separator ARIA role does not have the required ARIA attributes. (biome lint/a11y/useAriaPropsForRole)

However, according to the spec (https://www.w3.org/TR/wai-aria/#separator)
> There are two types of separators: a static structure that provides only a visible boundary and a focusable, interactive widget that is also moveable.

> If the separator is focusable, authors MUST set the value of aria-valuenow to a number...

https://biomejs.dev/playground/?code=PABkAGkAdgAgAHIAbwBsAGUAPQAiAHMAZQBwAGEAcgBhAHQAbwByACIAIAA%2BACAAfAAgADwALwBkAGkAdgA%2BAA%3D%3D

**Comment by ematipico** [comment]
Please @astanciu , next time don't delete the template. The template asked for "Expected Result".

It's not very clear, at least to me, what's the bug report about, and what it should be done to fix the issue (hence, the expected result).

**Comment by Shadowfiend** [comment]
Expectation-wise, providing a role="separator" without any additional attributes should pass this rule as well as lint/a11y/useFocusableInteractive on its parent element with no notices; this is because, as quoted above, an element with `role="separator"` and no additional attributes is a non-focusable separator, which requires no additional ARIA attributes.

**Comment by bluevoid-pl** [comment]
Here are more details that I believe are relevant to this issue @ematipico.
Given that separator is [Non focusable](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/separator_role#non-focusable_separator) it is correct to not pass any aria attributes(defaults should be enough).
Following error can be seen when using `role="separator"` tested on biome 2.2.5, file input-otp.tsx from shadcn/ui:
```
lib/components/ui/input-otp.tsx:69:40 lint/a11y/useAriaPropsForRole ━━

  ✖ The element with the separator ARIA role does not have the required ARIA attributes.
  
     67 │ function InputOTPSeparator({ ...props }: React.ComponentProps<"div">) {
     68 │        return (
   > 69 │                <div data-slot="input-otp-separator" role="separator" {...props}>
        │                                                     ^^^^^^^^^^^^^^^^
     70 │                        <IconMinus />
     71 │                </div>
  
  ℹ Missing ARIA prop(s):
  
  - aria-valuenow
 ```
Correct behavior should be no error, for `role="separator"`. Attribute `aria-valuenow` must be used when separator is focusable and value is known and it should not be set when separator is non focusable or value is unknown.
To improve this, biome should detect is element is focusable(is focusable tag or have tabIndex set) first and then display warning or info(preferred) that `aria-valuenow` may be needed if value is known.(which will probably not be true for most cases)
Relevant [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/separator_role#all_descendants_are_presentational) citation: 

> **[`aria-valuenow`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-valuenow)**
>    If the separator is focusable and has a known value, the `aria-valuenow` defines the current value. If not focusable or the value is unknown, do not include this attribute. 

PS. It seems that rule `lint/a11y/useFocusableInteractive` is also broken. It flags `role="separator"` for being focusable, even when there is no reason to think that it is interactive.
